repos/system_upgrade/common/actors/checkldsoconf/libraries/checkldsoconfiguration.py

```diff
@@ -0,0 +1,123 @@
+import glob
+import os
+
+from leapp import reporting
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api, CalledProcessError, run
+from leapp.models import InstalledRedHatSignedRPM
+
+LD_SO_CONF_DIR = '/etc/ld.so.conf.d'
+LD_SO_CONF_MAIN = '/etc/ld.so.conf'
+LD_SO_CONF_DEFAULT_INCLUDE = 'ld.so.conf.d/*.conf'
+LD_SO_CONF_COMMENT_PREFIX = '#'
+FMT_LIST_SEPARATOR = '\n    - '
+
+
+def _read_file(file_path):
+    with open(file_path, 'r') as fd:
+        return fd.readlines()
+
+
+def _report_custom_ld_so_configuration(summary):
+    reporting.create_report([
+        reporting.Title(
+            'Third party libraries linked with ld.so.conf are not supported.'
+        ),
+        reporting.Summary(summary),
+        reporting.Remediation(hint=('Remove the custom ld.so configuration and apply the changes '
+                                    'using the ldconfig command.')),
+        reporting.RelatedResource('file', '/etc/ld.so.conf'),
+        reporting.RelatedResource('directory', '/etc/ld.so.conf.d'),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Groups([reporting.Groups.OS_FACTS]),
+    ])
+
+
+def _is_modified(config_path):
+    """ Decide if the configuration file was modified based on the package it belongs to. """
+    result = run(['rpm', '-Vf', config_path], checked=False)
+    if result['exit_code'] == 0:
+        return False
+    output = result['stdout']
+    # Note that the condition is relaxed so that files different only
+    # in the modification timestamp won't be considered as modified.
+    modification_flags = output.split(' ', 1)[0]
+    return modification_flags.strip('.') != 'T'
```

Comment:
@PeterMocary  Reading the comment, it seems to me the only problem you are trying to resolve by the check of the output is whether the file contains some changes (hash, chmod, ...) or whether the timestamp has been changed only. correct? if so, this could simplify the solution:

```suggestion
    result = run(['rpm', '-Vf', '--nomtime', config_path], checked=False)
    if result['exit_code'] == 0:
        return False
    return True
```
This will ignore change of timestamp and report any other changes.

Reply:
I changed this to check only the checksum of the file as we discussed offline.